### ISSUE TITLE
[TCL] regsub/regexp -- and double quotes

### ISSUE THREAD

**Issue by dabro666** [sublimehq/Packages#1409]
Sublime does not recognise open double quote after 'regsub -- ' or 'regexp -- ', so the close double quote is interpreted as open one and all highlightning does not work correctly after this construction.

**Comment by dabro666** [comment]
Easiest way to fix is to replace line 133 from `- match: \{` to  `- match: '(\{|")'`.
It's not correct at all, but it allows to work with code than without this fix

**Comment by wbond** [comment]
Can you provide a sample of code that is broken? I don't know enough TCL to know precisely what you mean.

Also, can you indicate what build of Sublime Text you are using?

**Comment by dabro666** [comment]
Build 3143 for all platforms

Here is example:
```tcl
foreach {one_arg_opt_pattern} [list {-first\S*} {-second\S*} {-group\S*}] {
    regsub -- "${one_arg_opt_pattern}\\s+\\S+" $args {} args
}
foreach {boolean_opt_pattern} [list {-default\S*}] {
    regsub -- "${boolean_opt_pattern}\\s*" $args {} args
}

#    puts "..args parsed:"
#    puts "_add: '$_add'"
#    puts "_name: '$_name'"
##    puts "_group: '$_group'"
##    puts "_src: '$_src'"
#    puts "args: '$args'"
```

And some screens:
Broken one (look at highlighted double quotes and comment's highlighting)
![1](https://user-images.githubusercontent.com/31410405/36089770-8553b4f8-0fee-11e8-93d1-5d29e4ea7ea1.png)

Not correct enough, but better looking with my solution
![2](https://user-images.githubusercontent.com/31410405/36089773-8b2f4ec8-0fee-11e8-9832-c074fe07f68a.png)

**Comment by kleinfreund** [comment]
This affects the current dev build as well.

**Comment by wbond** [comment]
This is fixed in 2b8971497668cb724db07cc67f676b61eef23853

**Comment by dabro666** [comment]
Thanks!
